### DISCUSSION TITLE
What am I doing wrong?

### DISCUSSION THREAD

**Discussion by Ottie93** [raomin/ESPAltherma#420] [Q&A] [answered]
Hello all,

Recently I have been learning about monitoring my Daikin heatpump. Via a forum I discovered ESPAltherma and started by setting up a Raspberry Pi with HomeAssistant. So far, so good. 

Then I had to install the M5StickC, that got a bit more complicated. Although the guide is quite handy, it didn't help me solve a problem: errors when uploading the code to the Stick. I did everything the guide said, but I still got these errors:

`src/main.cpp: In function 'void extraLoop()':
src/main.cpp:100:8: error: 'class M5StickC' has no member named 'Display'
     M5.Display.wakeup();
        ^~~~~~~
src/main.cpp:103:8: error: 'class M5StickC' has no member named 'Display'
     M5.Display.sleep();
        ^~~~~~~
src/main.cpp: In function 'void setupScreen()':
src/main.cpp:256:10: error: 'class M5Display' has no member named 'setBrightness'
   M5.Lcd.setBrightness(127);
          ^~~~~~~~~~~~~
src/main.cpp:258:10: error: 'class M5Display' has no member named 'setFont'; did you mean 'textfont'?
   M5.Lcd.setFont(&FreeSansBold12pt7b);
          ^~~~~~~
          textfont
src/main.cpp:266:10: error: 'class M5Display' has no member named 'setFont'; did you mean 'textfont'?
   M5.Lcd.setFont(&Font0);
          ^~~~~~~
          textfont
src/main.cpp:266:19: error: 'Font0' was not declared in this scope
   M5.Lcd.setFont(&Font0);
                   ^~~~~
src/main.cpp:266:19: note: suggested alternative: 'font'
   M5.Lcd.setFont(&Font0);
                   ^~~~~
                   font`

I didn't know what to do, since I never worked with Visual Code Studio before, but a friend of mine has and told me there was probably a library missing. So I searched for "M5StickC" in the PlatformIO library and installed the package into the ESPaltherma project. I then uploaded the code to the Stick again and then it worked, although I haven't installed the Stick to the heatpump yet. 

So my question is: did I do it right? I got one entity (AlthermaSensors) visible in HA, but in the guide it states there should be 2 (switch.altherma is missing), but maybe that's because I haven't installed it on the heatpump yet?

**Comment by raomin** [comment, marked as answer]
This should have been fixed a week ago. Did you try with recent code?

**Reply by Ottie93** [reply]
I was indeed using the code from 2 weeks ago and you are right, it works now. Thank you.
